File: doordeck-sdk/src/mingwMain/resources/doordeck_headless_sdk/Model/Responses/FusionResponses.cs
using System.Text.Json.Serialization;

namespace Doordeck.Headless.Sdk.Model.Responses
{
    public class FusionLoginResponse
    {
        public string AuthToken { get; set; } = string.Empty;
    }
    
    public class IntegrationTypeResponse
    {
        public string Status { get; set; } = string.Empty;
    }
    
    public class DoorStateResponse
    {
        public ServiceStateType State { get; set; } = ServiceStateType.UNDEFINED;
    }
    
    public class IntegrationConfigurationResponse
    {
        public ControllerResponse? Doordeck { get; set; } = null;
        public ServiceStateResponse? Service { get; set; } = null;
        public DiscoveredDeviceResponse? Integration { get; set; } = null;
    }
    
    public class ControllerResponse
    {
        public string Id { get; set; } = string.Empty;
        public string? Name { get; set; } = null;
        public string? Role { get; set; } = null;
    }
    
    public class ServiceStateResponse
    {
        public ServiceStateType State { get; set; } = ServiceStateType.UNDEFINED;
    }
    
    public class DiscoveredDeviceResponse
    {
        public ILockController Key { get; set; } = new DemoController();
        public Dictionary<string, string> Metadata { get; set; } = new();
    }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum ServiceStateType
    {
        RUNNING,
        STOPPED,
        UNDEFINED
    }
}

File: doordeck-sdk/src/commonMain/kotlin/com/doordeck/multiplatform/sdk/model/responses/FusionResponses.kt
package com.doordeck.multiplatform.sdk.model.responses

import com.doordeck.multiplatform.sdk.model.common.ServiceStateType
import com.doordeck.multiplatform.sdk.model.common.UserRole
import com.doordeck.multiplatform.sdk.model.data.BasicLockController
import kotlinx.serialization.Serializable

internal typealias BasicLockControllerResponse = BasicLockController

@Serializable
internal data class BasicFusionLoginResponse(
    val authToken: String
)

@Serializable
internal data class BasicIntegrationTypeResponse(
    val status: String? = null
)

@Serializable
internal data class BasicDoorStateResponse(
    val state: ServiceStateType
)

@Serializable
internal data class BasicIntegrationConfigurationResponse(
    val doordeck: BasicControllerResponse? = null,
    val service: BasicServiceStateResponse? = null,
    val integration: BasicDiscoveredDeviceResponse? = null
)

@Serializable
internal data class BasicControllerResponse(
    val id: String,
    val name: String? = null,
    val role: UserRole? = null
)

@Serializable
internal data class BasicServiceStateResponse(
    val state: ServiceStateType
)

@Serializable
internal data class BasicDiscoveredDeviceResponse(
    val key: BasicLockControllerResponse,
    val metadata: Map<String, String>
)